### PR TITLE
Retry when pipe is busy on inheritable pipe creations

### PR DESCRIPTION
**Pull request by narasamdya**
The root cause of the bug is still unknown.

This change doesn't solve the bug, but try to address it by retry. If, after retry, the bug still occurs, then we will reach out to the Windows team.

[AB#1614763](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1614763)

### CODE REVIEW

File: Public/Src/Engine/Processes/Internal/Pipes.cs [hunk 1]
```diff
@@ -108,17 +108,48 @@ public static void CreateInheritablePipe(PipeInheritance inheritance, PipeFlags
...
-                pipeName,
-                FileDesiredAccess.GenericRead,
-                FileShare.None,
-                FileMode.Open,
-                readSideFlags,
-                out readHandle);
+            int maxRetry = 3;
 
-            if (!openReadSideResult.Succeeded)
+            while (true)
```
Comment (almiliMSFT) [inline Public/Src/Engine/Processes/Internal/Pipes.cs]: don't we have some generic utility method for retrying an arbitrary action?

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/Internal/Pipes.cs]: maybe not, or at least I couldn't find anything

Comment (Kr1sso) [reply Public/Src/Engine/Processes/Internal/Pipes.cs]: Actually `RetryOnFailure` from the IO helpers could be used here?

Comment (narasamdya) [reply Public/Src/Engine/Processes/Internal/Pipes.cs]: The retry logic here is a bit different from `RetryOnFailure`. We are doing wait for at most 5 sec in a loop, and then retry.

**Pull request merged**